_studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp

```diff
@@ -1171,8 +1190,14 @@ mfxStatus VideoDECODEAV1::FillOutputSurface(mfxFrameSurface1** surf_out, mfxFram
     UMC::VideoDataInfo const* vi = fd->GetInfo();
     MFX_CHECK(vi, MFX_ERR_DEVICE_FAILED);
 
+    surface_out->Data.TimeStamp = bs->TimeStamp != static_cast<mfxU64>(MFX_TIMESTAMP_UNKNOWN) ? bs->TimeStamp : GetMfxTimeStamp(m_frame_order * m_in_framerate);
```

Comment:
Seems using bitstream here is not good idea. It is better to store timestamp in the pFrame struct (somewhere in the umc_av1_* code) and then just use GetMfxTimeStamp() function here.

Reply:
Echo Ilya. It's better to set/get TS via pFrame . Two reasons:
1) Just to code style (we do this way for the rest decoders)
2) I assume AV1 has a re-ordering. If so, TS from bistream (in encoded order) can't be set to decoded surface (in display order).

Reply:
@wangyan42164 , everything except this part is fine to me. Can you please re-factor the item which Ilya pointed at? I understand that you borrowed the logic from VP9 but it wan't written a good way there.

Reply:
Sure. I have changed it in the latest commit. @dmitryermilov @dmitry-gurulev @IlyaSergeyev Could you please review it again? Thanks.